### PR TITLE
WIP Fixed custom serialization doco

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Related to https://github.com/Particular/docs.particular.net/issues/1020

 * Fixed the V6 API doco snippets on how to register a custom serializer
 * Added V6 version of `external-json` sample

TODO:
 * [x] Update reference to core
 * [x] Update reference to external json serializer

### CODE REVIEW

File: Snippets/Snippets_6/Snippets_6.csproj [hunk 1]
```diff
@@ -92,6 +92,7 @@
     </Reference>
     <Reference Include="NServiceBus.Core, Version=6.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
       <HintPath>..\..\packages\NServiceBus.6.0.0-unstable1423\lib\net452\NServiceBus.Core.dll</HintPath>
+      <HintPath>..\..\packages\NServiceBus.6.0.0-pullrequest3310\lib\net452\NServiceBus.Core.dll</HintPath>
```
Comment (hmemcpy) [inline Snippets/Snippets_6/Snippets_6.csproj]: Don't forget to replace this with a "proper" unsable build from Myget

File: Snippets/Snippets_6/packages.config [hunk 2]
```diff
@@ -8,7 +8,7 @@
   <package id="CommonServiceLocator" version="1.3" targetFramework="net452" />
   <package id="log4net" version="2.0.5" targetFramework="net452" />
   <package id="NLog" version="4.2.3" targetFramework="net452" />
-  <package id="NServiceBus" version="6.0.0-unstable1423" targetFramework="net452" />
+  <package id="NServiceBus" version="6.0.0-pullrequest3310" targetFramework="net452" />
```
Comment (hmemcpy) [inline Snippets/Snippets_6/packages.config]: Don't forget to replace this with a "proper" unsable build from Myget

**Comment by SimonCropp** [comment]
@SzymonPobiega marked as WIP till we fix the upstream stuff

**Comment by SzymonPobiega** [comment]
@SimonCropp :+1:

**Comment by danielmarbach** [comment]
Rebase?

**Comment by SimonCropp** [comment]
No point until we finish with the upstreams

**Comment by SimonCropp** [comment]
merged manually here https://github.com/Particular/docs.particular.net/commit/e3caca192d6ba6c076fc5691fc5ca9c97da44aa4

**Pull request closed without merging**